sdk/core/core/src/az_span.c

```diff
@@ -131,95 +175,128 @@ AZ_NODISCARD az_result az_span_to_uint32(az_span span, uint32_t* out_number)
 
 AZ_NODISCARD az_result az_span_copy(az_span destination, az_span source, az_span* out_span)
 {
-  AZ_PRECONDITION_VALID_SPAN(destination, 0, true);
-  AZ_PRECONDITION_VALID_SPAN(source, 0, true);
+  AZ_PRECONDITION_NOT_NULL(out_span);
+
   int32_t src_len = az_span_length(source);
+  int32_t dest_capacity = az_span_capacity(destination);
 
-  if (az_span_capacity(destination) < src_len)
+  if (dest_capacity < src_len)
   {
     return AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY;
-  };
+  }
 
   uint8_t* ptr = az_span_ptr(destination);
+  if (ptr == NULL)
+  {
+    *out_span = AZ_SPAN_NULL;
+  }
+  else
+  {
+    //TODO: What should be the behavior if source.ptr is NULL?
```

Comment:
memmove does not care about ptr == NULL. It would just not move any data. That's why we didn't consider it
We might want to add a comment about it